### ISSUE TITLE
Port patterns from 27_crypto.rule

### ISSUE THREAD

**Issue by dmajda** [openSUSE/scanny#32]
We need to port patterns form the `rules/27_crypto.rule` file from the [old scanner](http://gitorious.org/code-scanner/ror-sec-scanner/):

```
Desc: Cryptographic Issues

# impact        CWE identifier          regex
medium          CWE-331                 (srand|seed|urandom)
```

**Comment by dmajda** [comment]
Part of this pattern got ported (and enhanced) in [`random_numbers_check.rb`](https://github.com/openSUSE/scanny/blob/master/lib/scanny/checks/random_numbers_check.rb).

@thomasbiege: I have two questions for you:
1. What is the `seed` pattern trying to detect? A variable name? Method name? I'd like to make it a bit more specific.
2. What is the `urandom` pattern trying to detect? `/dev/urandom`? In that case I would port it to match literal strings containing this substring.

**Comment by ghost** [comment]
1. the method name, b/c by default the seed it ok and I sometimes see a seed used with lower entropy then the default seed.
2. sub-string is fine, the quality of urandom is not as good as the one from random.

**Comment by dmajda** [comment]
@Lte I'm looking at this code:

``` ruby
def pattern_urandom
  "StringLiteral<string *= /urandom/>"
end
```

Could you make it use a string instead of a regexp? We don't use any regexp capabilities here so the regexp is not really needed. There are some instances of this in other checks too, it would be great to fix them all.

**Comment by dmajda** [comment]
The [regexp issue](https://github.com/openSUSE/scanny/issues/32#issuecomment-7092085) was fixed by 4fe3bbb3772cc7e4481917cc5c7afedd42574a96.

The pattern is completely implemented in `RandomNumbersCheck`, closing.
